File: SQLCLR/11-CShrpSqlServerSp/CShrpSqlServerSp/ap_First.cs
using System;
using System.Data;
using System.Data.SqlClient;
using System.Data.SqlTypes;
using Microsoft.SqlServer.Server;


public partial class StoredProcedures
{
    [Microsoft.SqlServer.Server.SqlProcedure]

    public static void ap_First()
    {
        // Put your code here
        SqlContext.Pipe.Send("Hello world!\n");
    }


    [Microsoft.SqlServer.Server.SqlProcedure]
    // returning value from stored procedures
    public static System.Int32 ap_First_2()
    {
        // Put your code here
        SqlContext.Pipe.Send("Hello world!\n");
        return 0;
    }
};

File: SQLCLR/12-VbSqlServerSp/VbSqlServerSp/ap_First.vb
Imports System
Imports System.Data
Imports System.Data.SqlClient
Imports System.Data.SqlTypes
Imports Microsoft.SqlServer.Server


Partial Public Class StoredProcedures
    <Microsoft.SqlServer.Server.SqlProcedure()> _
    Public Shared Sub ap_FirstVB()
        ' Add your code here
        SqlContext.Pipe.Send("Hello world!\n")

    End Sub


    <Microsoft.SqlServer.Server.SqlProcedure()> _
    Public Shared Sub cp_EqTypeListVB()
        ' Add your code here
        SqlContext.Pipe.Send("Hello world!\n")
        Using conn As New SqlConnection("Context Connection=true")
            Dim cmd As New SqlCommand()
            cmd.Connection = conn
            cmd.CommandText = "select * from dbo.EqType"
            cmd.CommandType = CommandType.Text

            conn.Open()

            Dim rdr As SqlDataReader
            rdr = cmd.ExecuteReader()
            SqlContext.Pipe.Send(rdr)

            rdr.Close()
        End Using



    End Sub



End Class



